### PR TITLE
#311 keep archive on download

### LINKED ISSUE #510: Add the `--keep-archive` option to the `crowdin download` command
The `crowdin download` command currently removes the downloaded ZIP archive with translations after extracting.

But sometimes it would be helpful to keep this ZIP file to somehow post-process it. For example, use as an artifact in some CI/CD integrations.

I'm suggesting adding an optional `--keep-archive` option to the `crowdin download` command. When this option is specified, the CLI shouldn't remove the downloaded ZIP archive with translations.

The `-h` description for this option: `Do not remove the downloaded archive with translations after it's extracting`

This new option should be applied only to the `crowdin download` command and not to its subcommands.

### PR DESCRIPTION
**Pull request by yzerk**

### CODE REVIEW

File: src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java [hunk 1]
```diff
@@ -23,13 +23,16 @@ public class ParamsWithFiles extends ProjectParams {
...
+
     private List<String> labels;
 
     private List<String> excludedTargetLanguages;
 
-    public void setExportOptions(Boolean skipTranslatedOnly, Boolean skipUntranslatedFiles, Boolean exportApprovedOnly) {
+    public void setExportOptions(Boolean skipTranslatedOnly, Boolean skipUntranslatedFiles, Boolean keepArchive, Boolean exportApprovedOnly) {
         this.skipTranslatedOnly = skipTranslatedOnly;
         this.skipUntranslatedFiles = skipUntranslatedFiles;
+        this.keepArchive = keepArchive;
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java]: this is not an export option, it should be set separately

File: src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java [hunk 2]
```diff
@@ -344,10 +346,14 @@ private Pair<File, List<String>> download(BuildProjectTranslationRequest request
                 .removeStart(f.getAbsolutePath(), baseTempDir.getAbsolutePath() + Utils.PATH_SEPARATOR))
             .collect(Collectors.toList());
 
-        try {
-            files.deleteFile(downloadedZipArchive);
-        } catch (IOException e) {
-            out.println(ERROR.withIcon(String.format(RESOURCE_BUNDLE.getString("error.deleting_archive"), downloadedZipArchive)));
+        if (!keepArchive) {
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java:349]: can we have a Unit test for this case?

File: src/main/java/com/crowdin/cli/commands/picocli/DownloadSubcommand.java [hunk 3]
```diff
@@ -45,6 +45,9 @@ class DownloadSubcommand extends ActCommandWithFiles {
     @CommandLine.Option(names = {"--skip-untranslated-files"}, descriptionKey = "params.skipUntranslatedFiles")
     protected Boolean skipUntranslatedFiles;
 
+    @CommandLine.Option(names = {"--keep-archive"}, descriptionKey = "params.keepArchive")
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/picocli/DownloadSubcommand.java]: please put this after the `exportApprovedOnly` option

**Comment by andrii-bodnar** [comment]
the `package-lock.json` shouldn't be changed

File: src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java [hunk 4]
```diff
@@ -23,11 +23,13 @@ public class ParamsWithFiles extends ProjectParams {
...
     private Boolean exportApprovedOnly;
 
+    private Boolean keepArchive;
+
     private List<String> labels;
 
     private List<String> excludedTargetLanguages;
 
-    public void setExportOptions(Boolean skipTranslatedOnly, Boolean skipUntranslatedFiles, Boolean exportApprovedOnly) {
+    public void setExportOptions(Boolean skipTranslatedOnly, Boolean skipUntranslatedFiles, Boolean keepArchive, Boolean exportApprovedOnly) {
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java]: unused parameter `keepArchive`

Comment (yzerk) [reply src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java]: done

File: src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java [hunk 5]
```diff
@@ -23,6 +23,8 @@ public class ParamsWithFiles extends ProjectParams {
     private Boolean skipUntranslatedFiles;
     private Boolean exportApprovedOnly;
 
+    private Boolean keepArchive;
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java]: Still extra property :)

Comment (yzerk) [reply src/main/java/com/crowdin/cli/properties/ParamsWithFiles.java]: ah. my bad. fixed now

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**